client/gutenberg/extensions/related-posts/edit.jsx

```diff
@@ -3,122 +3,171 @@
 /**
  * External dependencies
  */
+import apiFetch from '@wordpress/api-fetch';
 import classNames from 'classnames';
 import { __ } from '@wordpress/i18n';
-import { Fragment } from '@wordpress/element';
+import { Component, Fragment } from '@wordpress/element';
 import { BlockAlignmentToolbar, BlockControls, InspectorControls } from '@wordpress/editor';
-import { moment } from '@wordpress/date';
 import { Button, PanelBody, RangeControl, ToggleControl, Toolbar } from '@wordpress/components';
+import { withSelect } from '@wordpress/data';
 
 /**
  * Internal dependencies
  */
 import { ALIGNMENT_OPTIONS, DEFAULT_POSTS, MAX_POSTS_TO_SHOW } from './constants';
 
-export default ( { attributes, className, setAttributes } ) => {
-	const {
-		align,
-		displayContext,
-		displayDate,
-		displayThumbnails,
-		postLayout,
-		postsToShow,
-	} = attributes;
-
-	const layoutControls = [
-		{
-			icon: 'grid-view',
-			title: __( 'Grid View', 'jetpack' ),
-			onClick: () => setAttributes( { postLayout: 'grid' } ),
-			isActive: postLayout === 'grid',
-		},
-		{
-			icon: 'list-view',
-			title: __( 'List View', 'jetpack' ),
-			onClick: () => setAttributes( { postLayout: 'list' } ),
-			isActive: postLayout === 'list',
-		},
-	];
-
-	const displayPosts = DEFAULT_POSTS.slice( 0, postsToShow );
-
-	return (
-		<Fragment>
-			<InspectorControls>
-				<PanelBody title={ __( 'Related Posts Settings', 'jetpack' ) }>
-					<ToggleControl
-						label={ __( 'Display thumbnails', 'jetpack' ) }
-						checked={ displayThumbnails }
-						onChange={ value => setAttributes( { displayThumbnails: value } ) }
-					/>
-					<ToggleControl
-						label={ __( 'Display date', 'jetpack' ) }
-						checked={ displayDate }
-						onChange={ value => setAttributes( { displayDate: value } ) }
-					/>
-					<ToggleControl
-						label={ __( 'Display context (category or tag)', 'jetpack' ) }
-						checked={ displayContext }
-						onChange={ value => setAttributes( { displayContext: value } ) }
-					/>
-					<RangeControl
-						label={ __( 'Number of posts', 'jetpack' ) }
-						value={ postsToShow }
-						onChange={ value =>
-							setAttributes( { postsToShow: Math.min( value, MAX_POSTS_TO_SHOW ) } )
-						}
-						min={ 1 }
-						max={ MAX_POSTS_TO_SHOW }
+class RelatedPostsEdit extends Component {
+	state = {
+		posts: [],
+		fetchingPosts: false,
+	};
+
+	componentDidMount() {
+		this.fetchPosts();
+	}
+
+	componentDidUpdate( prevProps ) {
+		if ( prevProps.isSaving && ! this.props.isSaving ) {
+			this.fetchPosts();
+		}
+	}
+
+	fetchPosts() {
+		const { postId } = this.props;
+		const { fetchingPosts } = this.state;
+
+		if ( ! postId || fetchingPosts ) {
+			return;
+		}
+
+		this.setState( {
+			fetchingPosts: true,
+		} );
+
+		apiFetch( {
+			path: '/jetpack/v4/site/posts/related?http_envelope=1&post_id=' + postId,
+		} ).then( response => {
+			this.setState( {
+				posts: response.posts,
+				fetchingPosts: false,
+			} );
+		} );
```

Comment:
What if there's a problem fetching? I [think `apiFetch` will reject](https://github.com/WordPress/gutenberg/blob/24c9674b2feb0c4e7daf7b409ceb65b32fe7c90c/packages/api-fetch/src/index.js#L46-L52), which we don't appear to handle.

Do we need to handle rejection? It looks like the placeholders will work and `fetchingPosts` will be stuck to `true`, but I'm wondering if these details are intentional.